include/blast/math/simd/SimdSize.hpp

```diff
@@ -14,24 +14,20 @@
 
 #pragma once
 
+#include <blast/math/simd/Simd.hpp>
+
 #include <cstdlib>
+#include <type_traits>
 
 
 namespace blast
 {
-    template <typename T>
-    struct SimdSize;
-
-
-    template <typename T>
-    struct SimdSize<T const> : SimdSize<T> {};
-
-
     /**
-     * @brief Number of elements in a SimdVec<T> object.
+     * @brief Size of a SIMD register conatining scalars of a given type.
      *
      * @tparam T scalar type
+     * @tparam Arch instruction set architecture
      */
-    template <typename T>
-    size_t constexpr SimdSize_v = SimdSize<T>::value;
-}
\ No newline at end of file
+    template <typename T, typename Arch = xsimd::default_arch>
+    std::size_t constexpr SimdSize_v = xsimd::batch<std::remove_cv_t<T>, Arch>::size;
```

Comment:
Isn't this what `SS` was doing before? Do we have some duplication perhaps?

Reply:
`SS` is just a shorthand to avoid long expressions like `SimdSize_v<double>` or `xsimd::batch<double, Arch>::size`. `SS` is a private implementation detail. `SimdSize_v` is supposed to be a shorter version of `xsimd::batch<double, Arch>::size`, it also hides the dependency on `xsimd`. We can replace it with something more modern and convenient, if we want, e.g.
```c++
template <typename T, typename Arch>
constexpr std::size_t simdSize(Arch arch)
{
    return xsimd::batch<T, Arch>::size;
}
```